packages/redux/src/actions.ts

```diff
@@ -41,11 +41,22 @@ export const setApi: PayloadActionCreator<DhType> = api => ({
   payload: api,
 });
 
-export const setWorkspace: PayloadActionCreator<Workspace> = workspace => ({
+export const setWorkspace: PayloadActionCreator<
+  CustomizableWorkspace
+> = workspace => ({
   type: SET_WORKSPACE,
   payload: workspace,
 });
 
+export const setDefaultWorkspaceSettings: PayloadActionCreator<
+  WorkspaceSettings
+> = settings => ({
+  type: SET_DEFAULT_WORKSPACE_SETTINGS,
+  payload: {
+    ...settings,
+  },
```

Comment:
Pretty sure you don't need the spread here:
```suggestion
  payload: settings,
```